tensortrade/strategies/tensorforce_trading_strategy.py

```diff
@@ -23,34 +23,29 @@
 
 from tensorforce.agents import Agent
 from tensorforce.execution import Runner
+from tensorforce.environments import Environment
 
 from tensortrade.strategies import TradingStrategy
 
 
 class TensorforceTradingStrategy(TradingStrategy):
     """A trading strategy capable of self tuning, training, and evaluating with Tensorforce."""
 
-    def __init__(self, environment: 'TradingEnvironment', agent_spec: Dict, network_spec: Dict, **kwargs):
+    def __init__(self, environment: TradingEnvironment, agent_spec: any, save_best_agent: bool = True, **kwargs):
         """
         Arguments:
             environment: A `TradingEnvironment` instance for the agent to trade within.
-            agent_spec: A specification dictionary for the `Tensorforce` agent.
-            network_spec: A specification dictionary for the `Tensorforce` agent's model network.
+            agent: A `Tensorforce` agent or agent specification.
+            save_best_agent (optional): The runner will automatically save the best agent
             kwargs (optional): Optional keyword arguments to adjust the strategy.
         """
-        self._environment = environment
-
-        self._max_episode_timesteps = kwargs.get('max_episode_timesteps', None)
-
-        self._agent_spec = agent_spec
-        self._network_spec = network_spec
-
-        self._agent = Agent.from_spec(spec=agent_spec,
-                                      kwargs=dict(network=network_spec,
-                                                  states=environment.states,
-                                                  actions=environment.actions))
-
-        self._runner = Runner(agent=self._agent, environment=environment)
+        self._max_episode_timesteps = kwargs.get('max_episode_timesteps', False)
+        
+        self._environment = Environment.create(environment='gym', level=environment, max_episode_timesteps=self._max_episode_timesteps)
+        
+        self._agent = Agent.create(agent=agent_spec, environment=self._environment)
+        
```

Comment:
#Agent initialization

agent.initialize()

Reply:
The runner will handle initialization iirc.